arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -96,6 +98,11 @@ exception_common:
   addi       s0, sp, XCPTCONTEXT_SIZE
   REGSTORE   s0, REG_X2(sp)       /* original SP      */
 
+#ifdef CONFIG_ARCH_FPU
+  mv         a0, sp
+  jal        x1, riscv_savefpu
+#endif
```

Comment:
let's move after line 111 to mirror the restore order.

Reply:
It's better to keep current location since a0 in line 111 was occupied by mcause.

Reply:
But why not move mcause just before line 112?

Reply:
Done